### ISSUE TITLE
YogaSMC

### ISSUE THREAD

**Issue by marianopela** [erinviegas/Lenovo-Ideapad-330S-14IKB#6]
Hi, didn't want to go off-topic on the other issue 

I noticed you started using YogaSMC, and that's good. The more, the merrier
Just wanted to inform you that ECRW is not needed on Ideapad devices, and also maybe we can use this issue (or open one on my repo, if you prefer) to post our findings and help the developer with testing (as I'm doing with my 10th get Ideapad s540)

Also @kasti0 is probably interested

**Comment by kasti0** [comment]
I discovered this .kext the first time today. I´m still not sure what extra functionality it gives me. the most important one I´m interested in would be Fan read/ control, but thats not working.
the PrefPane that comes with it is also not working for me, I opened an issue on their github repo.

**Comment by marianopela** [comment]
> the most important one I´m interested in would be Fan read/ control, but thats not working

That is only available on Thinkpads for now. It will probably be implemented in the future (as stated in https://github.com/zhen-zen/YogaSMC/projects/3) so I guess when-zen needs our help with testing

**Comment by erinviegas** [comment]
> doing with my 10th get Id

Yes, I too discovered this a little while ago. Got things like Airplane mode working. In conservation mode, the battery charges only to 62%. I removed the ECRW.

**Comment by erinviegas** [comment]
> Hi, didn't want to go off-topic on the other issue
> 
> I noticed you started using YogaSMC, and that's good. The more, the merrier
> Just wanted to inform you that ECRW is not needed on Ideapad devices, and also maybe we can use this issue (or open one on my repo, if you prefer) to post our findings and help the developer with testing (as I'm doing with my 10th get Ideapad s540)
> 
> Also @kasti0 is probably interested

Yes, I too discovered this a little while ago. Got things like Airplane mode working. In conservation mode, the battery charges only to 62%. I removed the ECRW.

**Comment by kasti0** [comment]
@marianopela regarding fan read/ control:
There's a tool for windows called NbFanControl and several Lenovo thinkpads and ideapads are already implemented. Unfortunately no config works for the 330s.
The Programm comes with handy tools where you can read the EC dumps and see if any register is linked to the fan speed, but I didn't make any progress with that.
I examined the whole EC register of my DSDT and excluded all values that are already used by known methods or not at all. I hope I can find the ec registers for the fan at some point.

**Comment by marianopela** [comment]
@kasti0 I can't really try that atm, as I don't have Windows installed. Let's wait and see what zhen-zen tells us when he responds to your issue, fingers crossed

**Comment by kasti0** [comment]
Do you know what the DYTC function is that's mentioned in the YogaSMC repo? @marianopela

**Comment by marianopela** [comment]
> Do you know what the DYTC function is that's mentioned in the YogaSMC repo? @marianopela

Sure, in my 540s BIOS it is referred to as `System Performance Mode` (Intelligent, Quiet, Performance), and it's basically a performance profile regarding the fan mode. Unfortunately it doesn't seem to be present in the 330s model

**Comment by kasti0** [comment]
The bios of the 330s is completely locked down, unfortunately. Only some basic functionality is available. 
I just recently modded the bios of my 330s (see the the other "issue" I opened in this repo). I'll have a look at the bios dump to see if that functionality is hidden somewhere in the bios.
I just was confused because DYTC was marked with a green mark in the YogaSMC repo for IdeaVPC.

**Comment by marianopela** [comment]
DYTC is present in the normal bios on my Ideapd s540, so I’m not so sure you could find anything. Still worth having a look though

**Comment by kasti0** [comment]
There's only a "Boot Performance Mode" which seems to be something different.![image](https://user-images.githubusercontent.com/49281774/102669111-fdd94180-418d-11eb-9d1c-cd385498b777.jpeg)
Really annoying that this BIOS is so locked down.
I still hope fan control can be done with YogaSMC. As of now I don't really understand how the kext and all of it's add-ons work.
